### ISSUE TITLE
Index out of range exception

### ISSUE THREAD

**Issue by sskodje** [XamlAnimatedGif/XamlAnimatedGif#19]
http://i.imgur.com/MMSvD9j.gif

This GIF crashes the library with System.IndexOutOfRangeException. 

It was resized and optimized with http://ezgif.com/optimize. The original that works is here: http://i.imgur.com/sPwgpLj.gif

It DOES fix the crash if you change CodeTable.Add like this: 

```
            public void Add(Sequence sequence)
            {
                if (_count == _table.Length) //my code
                    return;

                _table[_count++] = sequence;
                if ((_count & (_count - 1)) == 0 && _codeLength < MaxCodeLength)
                    _codeLength++;
            }
```

But i have no idea if it has any unintended consequences.

**Comment by thomaslevesque** [comment]
Thanks for the report, I'll look into it.

**Comment by thomaslevesque** [comment]
Your fix seems to work, but I have to check a few things to be sure it makes sense. I wrote that code a while ago, and the subtleties of LZW decoding are not very fresh in my mind ;)

**Comment by thomaslevesque** [comment]
OK, I understand a bit better now. In other GIF images I had looked at, when the code table was full, the next code in the stream was always the clear code, which is an indication to reset the code table. It's not the case here; the optimizer must have decided that it was more efficient to reuse the same codes rather than resetting the code table. The code didn't allow for that, hence the error.

So your fix is correct. That was easy to verify empirically, but at least now I understand _why_ it's correct ;). I'll publish a new release with the fix shortly.

Thanks!

**Comment by thomaslevesque** [comment]
Fixed in 1.0.1

I also fixed an issue where the decoding error wasn't properly reported via the `Error` event. Instead it was causing a NullReferenceException when trying to raise the event... oops!
